apps/web/utils/transactionMessages.ts

```diff
@@ -0,0 +1,21 @@
+export const getTxMessage = (transactionStatus: string | undefined) => {
+  let message = "";
+  switch (transactionStatus) {
+    case "idle":
+      message = "";
+      break;
+    case "waiting":
+      message = "Waiting for signature...";
+      break;
+    case "loading":
+      message = "Transaction in progress...";
+      break;
+    case "success":
+      message = "Approved";
+      break;
+    case "error":
```

Comment:
Will not always be User rejected the request, you need to pass the error to determines thhe error message

Reply:
And maybe extract the `parseErrorMessage` from useTransactionnotification to here

Reply:
thanks for the catch!
implemented!